specs/Menu.md

```diff
@@ -1,731 +1,672 @@
+### Sections and submenus
+
+```typescript
+const trigger = <button> Open menu </button>
+
+const menu = (
+  <Menu trigger={trigger}>
+    <MenuItem>Option 1</MenuItem>
+    <MenuSection title="Section">
+      <MenuItem>Section Option 1</MenuItem>
+      <MenuItem>Section Option 2</MenuItem>
+      <MenuItem>Section Option 3</MenuItem>
+    <MenuSection />
+    <SubMenu title="Open submenu">
+      <MenuItem>Option 1</MenuItem>
+      <MenuItem>Option 2</MenuItem>
+      <MenuItem>Option 3</MenuItem>
+    <SubMenu>
+  <Menu>
+)
+```
 
-##### Focus
+```html
+<!-- expected DOM output  -->
+<!-- TODO positioning -->
+<div role="menu">
+  <div role="menuitem" tabindex="0">Option 1</div>
+  <div role="separator"></div>
+  <div role="menuitem" tabindex="-1">Section Option 1</div>
+  <div role="menuitem" tabindex="-1">Section Option 2</div>
+  <div role="menuitem" tabindex="-1">Section Option 3</div>
+  <div role="separator"></div>
+  <!-- TODO submenu -->
+</div>
+```
 
-By default Focus should go to the tab stops and it's up to each menuItem to decide if it has tabindex=0. An example of a menuItem that shouldn't be focusable would be a Divider.
+### Standlone
 
-However for the menu we should provide a focus wrapper that is able to track focus using role="menuitem" to determine what item should get focused.
+```typescript
+const [open] = React.useState(false);
 
-The menu does not need to provide a native way to be controlled without getting focus directly. Like above, there should be a simple utility wrapper that could be written to implement this.
+const menu = (
+  <CustomSurface open={open}>
+    <MenuList>
+      <MenuItem>Option 1</MenuItem>
+      <MenuItem>Option 2</MenuItem>
+      <MenuItem>Option 3</MenuItem>
+    <MenuList>
+  <CustomSurface>
+)
+```
 
-##### On Menu Click
+```html
+<!-- expected DOM output  -->
+<div role="menu">
+  <div role="menuitem" tabindex="0">Option 1</div>
+  <div role="menuitem" tabindex="-1">Option 2</div>
+  <div role="menuitem" tabindex="-1">Option 3</div>
+</div>
+```
 
-Some implementations of menu have a general `onClick` function that is applied to all items as a way of getting more information.
+### Selection
+
+```typescript
+const trigger = <button> Open menu </button>
+const [selectedItems, setSelectedItems] = React.useState([]);
+
+// basic checkbox example
+const menuCheckbox = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuItem index={1}>Option 1</MenuItem>
+    <MenuItem index={2}>Option 2</MenuItem>
+    <MenuItem index={3}>Option 3</MenuItem>
+  <Menu>
+)
+
+// leverage MenuSection for different selection groups
+const menuSelectableSections = (
+  <Menu
+    selectedItems={selectedItems}
+    onSelectionChange={setSeelctedItems}
+    trigger={trigger}
+  >
+    <MenuSection kind="checkbox">
```

Comment:
What is this `MenuSection` good for?

Reply:
`MenuSection` would be responsible for adding dividers and section headers in a correct way in terms of html semantics and narration. It would also help manage create selectable groups of menu checkboxes and radios which require some extra DOM elements for a consisent narration experience

Reply:
I see the additional wrapping element for radiogroup, I just wasn't sure about the checkbox section.

Reply:
Ah that's a mistake on my part will fix, my intention is that any selectable group in a `MenuSection` needs to be wrapped to provide a consistent narration experience. 

This also means that if you have a menu with only checkboxes it actually makes sense **not** to use `MenuSection`

Reply:
Added the missing group